algosdk/future/transaction.py

```diff
@@ -216,7 +216,10 @@ def undictify(d):
             txn = PaymentTxn(**args)
         elif txn_type == constants.keyreg_txn:
             args.update(KeyregTxn._undictify(d))
-            txn = KeyregTxn(**args)
+            if "nonpart" in d and d["nonpart"]:
```

Comment:
Since the recommendation is to not create a new `txn_type`, it seems like the only way to distinguish the transactions is through the `nonpart` parameter, which we do here. I'm not sure if there's a better solution.

Reply:
This looks good to me